chatbot/middleware/rate_limiting.py

```
@@ -0,0 +1,368 @@
+        }
+        
+        for endpoint, limit_type in endpoint_mappings.items():
+            if endpoint in path:
+                return limit_type
+        
+        return None
+    
+    def _check_rate_limit(
+        self, 
+        cache_key: str, 
+        limit_config: Dict, 
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)